src/browser/api_protocol/transport_strategy/elipc_strategy.ts

```diff
@@ -168,28 +235,23 @@ export class ElipcStrategy extends ApiTransportBase<MessagePackage> {
         };
     }
 
-    private addBreadcrumb(message: any,  messageId: number, action: string, name: string): void {
-        if (message && message.breadcrumbs && message.breadcrumbs.length) {
-            message.breadcrumbs.push({
-                action,
-                messageId,
-                name,
-                time: Date.now()
-            });
-        }
-    }
-
-    protected ackDecorator(e: any, messageId: number, originalPayload: any): AckFunc {
-        const ackObj = new AckMessage();
+    protected createAckObject(configuration: ElIPCConfiguration, originalPayload: any, messageId: number): AckMessage {
+        const usingBreadcrumbs = configuration.breadcrumbConfiguration.enabled;
+        const ackObj = (!usingBreadcrumbs ?
+                            new AckMessage()
+                                :
+                            new AckMessage(originalPayload.breadcrumb, originalPayload.action));
         ackObj.correlationId = messageId;
+        return ackObj;
+    }
 
-        // It is inferred that breadcrumbs have been enabled for a window
-        // when the breadcrumb array is present in the original payload of the API request.
-        const usingBreadcrumbs = originalPayload.breadcrumbs && originalPayload.breadcrumbs.length;
+    protected ackDecorator(e: any, messageId: number, originalPayload: any, baseConfiguration: MessageConfiguration): AckFunc {
+        const configuration: ElIPCConfiguration = <ElIPCConfiguration>baseConfiguration;
```

Comment:
I'm confused, casting to `ElIPCConfiguration`, can't it be defined in the params as `ElIPCConfiguration` instead of `MessageConfiguration`?

Same question for line 281

Reply:
This was recommended by @pbaize to avoid swaths of refactoring to the entire transport layer. ApiTransportBase<T> is providing the method signature and further coupling the configurations at the base-class for el_ipc specific fields was avoided.